pyproject.toml

```diff
@@ -1,7 +1,58 @@
+[project]
+name = "matplotlib"
+description = "Python plotting package"
+readme = "README.md"
+license = { file = "LICENSE/LICENSE" }
+dynamic = ["authors", "version"]
+classifiers=[
+    'Development Status :: 5 - Production/Stable',
+    'Framework :: Matplotlib',
+    'Intended Audience :: Science/Research',
+    'Intended Audience :: Education',
+    'License :: OSI Approved :: Python Software Foundation License',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Topic :: Scientific/Engineering :: Visualization',
+]
+dependencies = [
+    "contourpy>=1.0.1",
+    "cycler>=0.10",
+    "fonttools>=4.22.0",
+    "kiwisolver>=1.0.1",
+    "numpy>=1.19",
+    "packaging>=20.0",
+    "pillow>=6.2.0",
+    "pyparsing>=2.2.1",
+    "python-dateutil>=2.7",
+    "setuptools_scm>=7.0"
```

Comment:
This should be conditional, but I have not found any info how to do this dynamically.

Reply:
I don't think it is needed at all as it is in the `[build-system] requires` section. It is needed by anyone who uses `--no-build-isolation` in their `pip install .`, but in that situation they will need to have preinstalled all of the `[build-system] requires` packages to proceed.

Although I might be missing some other use of `setuptools_scm` in testing or somewhere else?

Reply:
It was added as someone had problems running tests in a particular setup(?). 

Without it, this breaks in the tests:
https://github.com/matplotlib/matplotlib/blob/007c4c126193aaa2d691695d6ccadb804fc598c8/lib/matplotlib/__init__.py#L171-L180

although it indeed is not really required for a user install.